### DISCUSSION TITLE
Noob question - can you explain git / iCloud drive / dropbox to me?

### DISCUSSION THREAD

**Discussion by pessinafederico** [glushchenko/fsnotes#1513] [General]
Hi everyone! Loving the app but I'm a bit confused from the versioning options

I have set up iCloud drive as my default location, which I need in order to have my notes across my devices. 

I would like to have a backup of all my notes, with versioning history. To do this, I did set up the git-backup preferences, which does a backup every 0 hours 10 minutes automatically to a folder within "Application Support". 

I would like thought for this backup to be saved on Dropbox - i.e. I don't want this folder to be saved only locally on my computer (I believe "application support" folder is only local) but I would like to have the extra security given by Dropbox. 

How can I achieve this?
1) Tell dropbox to backup this folder too?
2) Create a new local git repo within dropbox?

If the answer is 2), could you provide me some guidance on how to do it, because I didn't really understand the guide. 

Apologies for the extremely noob question !

**Comment by gingerbeardman** [comment]
From: https://github.com/glushchenko/fsnotes/wiki/Backup-&-Versioning-of-Notes

> The (git) repositories ... are not recommended for storage in the cloud.

Aside, if you have Time Machine on then all note will be backed up (open a note in Text Edit and choose Revert To > Browse All Versions...

**Comment by glushchenko** [comment]
Hi! You should not store .git repositories in Dropbox or iCloud Drive, because cloud providers are very bad at syncing the tens of thousands of files that git creates.

Instead, you must set up a repository to which you want to push backups. This can be done on github or any other git storage.

Guide about git push here:

https://github.com/glushchenko/fsnotes/wiki/Git-configuration

**Comment by pessinafederico** [comment]
Ok understood, thanks a lot for the explanation. I think I managed to make it work! Thanks! I hope you're well in Ukraine, wish you the best.
